incubator/command-log/src/main/java/io/aklivity/zilla/runtime/command/log/internal/LoggableStream.java

```diff
@@ -398,6 +406,25 @@ private void onData(
         }
     }
 
+    private void onDataPayload(
+        final DataFW data)
+    {
+        final int offset = data.offset() - HEADER_LENGTH;
+        final long timestamp = data.timestamp();
+        final OctetsFW payload = data.payload();
+
+        if (payload != null)
+        {
+            StringBuilder hexData = new StringBuilder();
+            for (int i = 0; i < data.length(); i++)
+            {
+                hexData.append(String.format("%02x", payload.buffer().getByte(payload.offset() + i)));
+                hexData.append(i < data.length() - 1 ? ':' : "");
+            }
```

Comment:
Consider `BitUtil.toHex(...)` instead.

Reply:
I tried but it didn't work. `BitUtil.toHex` needs `byte[]` as argument. I can provide `payload.buffer().byteArray()` which compiles, but it returns `null` so it throws a NPE in runtime.